### ISSUE TITLE
Account using too much memory leading to access violation in stack

### ISSUE THREAD

**Issue by vivaviche** [project-serum/anchor#708]
Debugged with https://github.com/xjcaa today

https://github.com/project-serum/anchor/blob/master/tests/escrow/programs/escrow/src/lib.rs

```
pub struct Exchange<'info> {
....

pub just_one_more_account Account<'info, TokenAccount>
}
```

This would lead to access violation in stack.

The current suspicion is that the new Account implementation is using too much memory.

If you change it from Account to AccountInfo, it will work. So the below works

```
pub struct Exchange<'info> {
....

pub just_one_more_account AccountInfo<'info>
}
```

**Comment by armaniferrante** [comment]
There are three options here.

1. Use references inside `Account` instead of cloning the AccountInfo.
2. Use Box<Account<...>>, which willhelp  mitigate access violations.
3. Use the zero copy Loader instead of `Account`.

1. is being tracked here https://github.com/project-serum/anchor/issues/512. Closing in favor of the original issue.

**Comment by vivaviche** [comment]
@armaniferrante Quick Q with regards to 1.

In this case, we see `Access Violation` BEFORE we can even get into the processor. So I suspect that it is taking too much memory even way before we call ".clone()".

Does this information mean anything with regards to the solutions?

**Comment by armaniferrante** [comment]
> @armaniferrante Quick Q with regards to 1.
> 
> In this case, we see `Access Violation` BEFORE we can even get into the processor. So I suspect that it is taking too much memory even way before we call ".clone()".
> 
> Does this information mean anything with regards to the solutions?

No that's expected. The `.clone()` I'm referring to is done inside account deserialization, which is before the processor is invoked.

**Comment by huybuidac** [comment]
@armaniferrante can you explain more detail about Box<Account<>>?, when should we use?

Anchor doesn't have clearly doc :(
